src/main/java/org/apache/mesos/offer/OfferEvaluator.java

```diff
@@ -317,16 +305,17 @@ private FulfilledRequirement(
         public List<OfferRecommendation> getCreateRecommendations() {
             return createRecommendations;
         }
+
```

Comment:
ResourceRequirement is immutable? there is non-final object there already. I did not see any comment saying this should be immutable :)